### PR TITLE
Sampling from a multinomial distribution method

### PR DESCRIPTION
**Pull request by xabomon**
This PR aims to add a method to wave_function to sample the state from a multinomial distribution rather than using choice, aligned with #2917 

Assistance needed to link the new function to the simulator and all other data frameworks of Cirq.

This PR is at a primary stage (WIP) and with pending tests and other CI changes.

### CODE REVIEW

File: cirq/sim/wave_function.py [hunk 1]
```diff
@@ -243,7 +244,66 @@ def sample_state_vector(
...
+    # Calculate the measurement probabilities.
+    probs = _probs(state, indices, qid_shape)
+
+    # We now have the probability vector, correctly ordered, so sample over
+    # it. Note that we use multinomial that returns how many times a value
+    # is repeated out of repetitions.
+    result = prng.multinomial(n=repetitions, pvals=probs, size=None)
+
+    # Convert to individual qudit measurements.
+    meas_shape = tuple(qid_shape[i] for i in indices)
```
Comment (obriente) [inline cirq/sim/wave_function.py:305]: This line doesn't seem to do anything, is something missing or should it be removed?

**Comment by obriente** [comment]
What's the status here? I just ran into this issue again - found a speedup of ~1000 from putting together a hack myself that uses multinomial.

Looking at the code, it appears it needs a) tests, and b) some wrapping to be generally accessible - anything else?

**Comment by xabomon** [comment]
Hi Tom,

Indeed, testing is needed.
What do you mean by wrapping to be generally accessible?

This functionality does not solve the issue of having to run multiple circuits for each commuting or qubit-wise commuting group. It is something we discussed at some OF sync.

**Comment by balopat** [comment]
Hi @xabomon, do you plan working on this PR? If not, I think we can close it. Thank you for the preliminary implementation, someone else can take it from here.

**Comment by xabomon** [comment]
I'm going to close the issue.

Xavi

**Pull request closed without merging**